frontend/src/app/cities-upload/cities-upload.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component } from '@angular/core';
+import { SearchCityService } from '../search-city.service';
+import { HttpEventType } from '@angular/common/http';
+import { Observable } from "rxjs/Observable";
+import 'rxjs/add/observable/interval';
+import 'rxjs/add/operator/finally';
+import 'rxjs/add/operator/take';
+import 'rxjs/add/operator/takeWhile';
+
+const ESTIMATED_PROCESSING_TIME_IN_MILLIS = 20000;
```

Comment:
AFAIK, there's nothing we can't do if we don't make the processing asynchronous: return a response immediately after the file is uploaded (but before it's processed), and the polling to see if the processing is done, or use SSE to get updates about the process. Not undoable, and probably somthing usefull if this was a core use-case, but since we don't even need this use-case here, and it's supposed to be used just once, I didn't want to go this way. Might be cool to experiment in a branch, though.